inst/pages/beta_diversity.qmd

```diff
@@ -236,6 +236,135 @@ tse <- runMDS(
 plotReducedDim(tse, "Unifrac", colour_by = "Group")
 ```
 
+### Rarefaction to mitigate impacts of uneven sequencing effort
+
+The sequencing depth of a sample refers to the number of gene sequences read 
+during the sequencing process. A variation in sequencing depth across the 
+samples of a study may impact the calculation of alpha and beta diversity 
+metrics [@Schloss2023]. 
+It is common to find significant variation in sequencing depth between samples.
+For instance, the samples of the *TreeSummarizedExperiment* dataset 
+`GlobalPatterns` show up to a 40-fold difference in the number of gene sequences.
+
+```{r rarefaction_1}
+# Calculate the list of sequencing depths across samples
+sequencing_depths <- colSums(assay(tse))
+# Calculate variation between highest and lowest sequencing depth 
+depth_variation <- max(sequencing_depths)/min(sequencing_depths)
+depth_variation
+```
+
+To address uneven sequencing effort, rarefaction aims to normalize sequence 
+counts using subsampling. 
+First, the user chooses the rarefaction depth and a number of iterations N. All the samples with sequence count below the rarefaction depth are removed and gene sequences are randomly drawn from the samples left to get subsamples fitting the rarefaction depth. Then a beta diversity metric is calculated from those subsamples and the process is iterated N times. Finally, beta diversity is estimated with the mean of all beta diversity values calculated on subsampled data.
+
+There has been a long-lasting controversy surrounding the use of rarefaction in
+microbial ecology. The main concern is that rarefaction would omit data. 
+However, if the subsampling process is repeated a sufficient number of times, 
+and if the rarefaction depth is set to the lowest sequence count found across all samples, no data will be omitted. 
+Moreover, Patrick Schloss has demonstrated that rarefaction is "the only method that could control for variation in uneven sequencing effort when measuring commonly used alpha and beta diversity metrics" [@Schloss2023].
+
+Before applying rarefaction, selecting the most variable features can help minimize variation caused by random subsampling. These features have the highest read counts, while rare features tend to increase sampling variation. This approach facilitates comparison of results between non-rarefied and rarefied distance calculations later on.  
+
+```{r, select highest variance features}
+tse0 <- GlobalPatterns
+
+# Let us see what happens when we operate with ntop highest variance features
+ntop <- 5
+tse0 <- tse0[head(rev(order(rowSds(assay(tse0, "counts")))), ntop), ]
```

Comment:
Is this filtering step needed?

Reply:
With this step, I can compare the rarefied result with the non-rarefied one:
```
print(all.equal(current = cor(as.vector(reducedDim(tse0, "MDS_aitchison")),
                        as.vector(reducedDim(tse0, "MDS_aitchison_rarefied"))),
          target = 1,
          tolerance = 0.05))
```
Without selecting the high variance features, the results are not correlated if I understood your previous comment about high variance features and correlation correctly.